Assets/Scripts/UI/PauseMenu.cs

```diff
@@ -0,0 +1,47 @@
+﻿using TMPro;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace UI
+{
+    public class PauseMenu : Menu
+    {
+        [SerializeField] private GameObject pausePanel;
+
+        // Update is called once per frame
+        private void Update()
+        {
+            PauseGame();
+        }
+
+        public void SetHighScoreText()
+        {
+            var highScores = ScoreManager.GetScores();
+            var scoreTextBoxes = highScoreTextPanel.GetComponentsInChildren<TextMeshProUGUI>();
+            
+            SetHighScoreText(highScores, scoreTextBoxes);
+        }
+        
+        private void PauseGame()
+        {
+            if (Input.GetKeyDown(KeyCode.Escape))
+            {
+                Time.timeScale = 0;
+                pausePanel.SetActive(true);
+            }
```

Comment:
Could be nice to be able to unpause with the escape key too

```suggestion
            if (Input.GetKeyDown(KeyCode.Escape))
            {
                Time.timeScale = 1 - Time.timeScale; // If scale is 1 -> 1 - 1 = 0, if scale is 0 -> 1 - 0 = 1
                pausePanel.SetActive(!pausePanel.active); 
            }
```
This way, each time we press escape it's gonna either pause or unpause the game, depending on it's current pause status
(the function could be renamed UpdatePauseStatus or smth like that, since it's purpose has changed)

Reply:
Good idea, I didn't think of this!